packages/plugin-typeorm/src/typeorm.ts

```diff
@@ -1,31 +1,60 @@
-import { BasePlugin } from '@opentelemetry/core';
-import { Span, SpanKind, StatusCode, setSpan, context } from '@opentelemetry/api';
+import { Span, SpanKind, StatusCode, setSpan, context, Logger } from '@opentelemetry/api';
 import { DatabaseAttribute, GeneralAttribute } from '@opentelemetry/semantic-conventions';
 import { TypeormPluginConfig } from './types';
-import { safeExecute, getParamNames } from './utils';
-import shimmer from 'shimmer';
+import { getParamNames } from './utils';
 import { VERSION } from './version';
-import * as typeorm from 'typeorm';
+import type * as typeorm from 'typeorm';
+import {
+    InstrumentationBase,
+    InstrumentationConfig,
+    InstrumentationModuleDefinition,
+    InstrumentationNodeModuleDefinition,
+    isWrapped,
+    safeExecuteInTheMiddle,
+} from '@opentelemetry/instrumentation';
+
+type Config = InstrumentationConfig & TypeormPluginConfig;
+
+export class TypeormInstrumentation extends InstrumentationBase<typeof typeorm> {
+    static readonly component = 'typeorm';
+    protected _config!: Config;
+
+    constructor(config: Config = {}) {
+        super('opentelemetry-instrumentation-typeorm', VERSION, Object.assign({}, config));
```

Comment:
What is the motivation in coping the config?
Do we modify it?

Reply:
You mean the Object.assign? I copied it from the official plugins but I believe it's a good practice